sdk/src/Core/RegionEndpoint/RegionEndpoint.cs

```diff
@@ -207,7 +213,81 @@ public static RegionEndpoint GetBySystemName(string systemName)
         /// <returns></returns>
         public static RegionEndpoint GetRegionEndpointOverride(RegionEndpoint regionEndpoint)
         {
-            return _hashRegionEndpointOverride.ContainsKey(regionEndpoint) ? _hashRegionEndpointOverride[regionEndpoint] : null;
+            lock (_hashRegionEndpointOverride)
```

Comment:
We should use a `ReaderWriterLockSlim` lock so we are obtaining an exclusive lock for every SigV4 sign request. Here is a recent example in the SDK using the `ReaderWriterLockSlim`. https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Credentials/DefaultInstanceProfileAWSCredentials.cs#L33